src/prototype_creep.js

```diff
@@ -63,7 +63,7 @@ Creep.prototype.handle = function() {
     return;
   }
 
-  if (this.memory.recycle) {
+  if (this.memory.recycle && this.memory.moveToSpawn) {
```

Comment:
do not like it at all. `creep.memory.recycle` meant to be universal and easily to set manually from interface, if we need it